File: src/main/java/apps_spring/aop/metier_spring/SecurityContext.java
package apps_spring.aop.metier_spring;

/**
 * to Simulate a Security context 
 * that can used as a service of security in AutorizationAspect
 */
public class SecurityContext {
    private static String username = "";
    private static String password = "";
    private static String[] roles = {};

    public static void authenticate(String usernameArg, String passwordArg, String[] rolesArg) {
        if (usernameArg.equals("root") && passwordArg.equals("1234")) {
            username = usernameArg;
            password = passwordArg;
            roles = rolesArg;
        } else {
            throw new RuntimeException("Access Denied!");
        }
    }

    public static boolean hasRoles(String roleArg) {
        for (String role : roles) {
            if (role.equals(roleArg)) return true;
        }

        return false;
    }
}

File: src/main/kotlin/applications_spring_kt/aop/metier/context/SecurityContext.kt
package applications_spring_kt.aop.metier.context

import kotlin.jvm.Throws

/**
 * a context to simulate a security context that can be used
 * as a service of security in AuthorizationAspect
 */
object SecurityContext {
    var username = "";
    var password = "";
    var roles: Array<String> = arrayOf("");

    @Throws(RuntimeException::class)
    fun authenticate(usernameArg: String = "", passwordArg: String = "", rolesArgs: Array<String> = arrayOf("")) {
        if (usernameArg.equals("root") && passwordArg.equals("1234")) {
            username = usernameArg;
            password = passwordArg;
            roles = rolesArgs;
        } else {
            throw RuntimeException("Access Denied !");
        }
    }

    fun hasRoles(roleArg: String = ""): Boolean {
        var result = false;
        roles.forEach {
            if (it.equals(roleArg)) {
                result = true;
                return@forEach
            }
        }

        return result;
    }


}